torch/_dynamo/variables/functions.py

```diff
@@ -771,6 +771,12 @@ def call_method(
                 grid=args[0],
                 **VariableTracker.propagate(self),
             )
+        elif name == "run":
+            if "grid" not in kwargs:
+                raise Unsupported("Triton kernel requires to be called with a grid")
+            grid = kwargs.pop("grid")
+            self.grid = grid
```

Comment:
VariableTrackers are read-only and should not be mutated.  You can do `self.clone(grid=grid)` to get a copy of `self` with a change applied to it.